bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,247 @@
+    private static List<FerroampChannelConfiguration> channelConfigSsoS1 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS2 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS3 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigSsoS4 = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEso = new ArrayList<>();
+    private static List<FerroampChannelConfiguration> channelConfigEsm = new ArrayList<>();
+
+    long refreshInterval = 30;
+    static boolean isEsoAvailable = false;
+    static boolean isEsmAvailable = false;
+
+    public FerroampHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String transId = UUID.randomUUID().toString();
+        String valueConfiguration = command.toString();
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonDisCharge, ferroampConfig);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            FerroampMqttCommunication.sendPublishedTopic(requestCmdJsonAuto, ferroampConfig);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigEhub = FerroampChannelConfiguration.getChannelConfigurationEhub();
+        channelConfigSsoS1 = FerroampChannelConfiguration.getChannelConfigurationSsoS1();
+        channelConfigSsoS2 = FerroampChannelConfiguration.getChannelConfigurationSsoS2();
+        channelConfigSsoS3 = FerroampChannelConfiguration.getChannelConfigurationSsoS3();
+        channelConfigSsoS4 = FerroampChannelConfiguration.getChannelConfigurationSsoS4();
+        channelConfigEso = FerroampChannelConfiguration.getChannelConfigurationEso();
+        channelConfigEsm = FerroampChannelConfiguration.getChannelConfigurationEsm();
+
+        final MqttBrokerConnection ferroampConnection = new MqttBrokerConnection(ferroampConfig.hostName,
+                FerroampBindingConstants.BROKER_PORT, false, false, ferroampConfig.userName);
+
```

Comment:
Before exiting this method you should set the thing state to unknown.
The async pollTask should later on set it to offline or online.

Reply:
20241125
Changes done to method initialize() in FerroampHandler.java